### DISCUSSION TITLE
Investigating CTSM5.2 surface datasets

### DISCUSSION THREAD

**Discussion by wwieder** [ESCOMP/CTSM#1868] [Show and tell]
New surface datasets have a number of changes that we can start discussing. I've created [this slide deck](https://docs.google.com/presentation/d/1fmReSb6XKQjChSjMTAxAp-p0BAR4HibsHya1y43udeg/edit?usp=sharing) to start illustrating changes between cases that we're running.  So far, these are i2000 CTSMsp simulations with 5.2 vs. 5.1 using GSWP3 forcing.  Both from a common CTSM5.1_dev090 tag.  [Diagnostics from the SP case are posted here](https://webext.cgd.ucar.edu/I2000/i2000sp_new_fsurdat/lnd/i2000sp_new_fsurdat.2_21-i2000sp.2_21/setsIndex.html) with associated raw data here `/glade/u/home/slevis/cases_sp`

- [ ] @lawrencepj1 do you have a script that can quickly compare differences in surface dataset fields?  If not, maybe @slevisconsulting can help?  These include: 
   - pft distributions, 
   - special land units like lakes, glacier, and urban 
   - soil properties (ORGANIC, CLAY and SAND + maybe the derived constants that getting used in the model like WATSAT or HKSAT that are written out to history files). 
- [x] Land over fractions, [especially for forests / grasses, see the PFT fractions](https://webext.cgd.ucar.edu/I2000/i2000sp_new_fsurdat/lnd/i2000sp_new_fsurdat.2_21-i2000sp.2_21/set2/set2.html) (e.g., forests in China ). @lawrencepj1 can you check these changes are as intended?
- [x] Lake, urban, and land ice datasets have changed for this 2000 case. The biggest thing that stands out here is the [distribution of lakes](https://webext.cgd.ucar.edu/I2000/i2000sp_new_fsurdat/lnd/i2000sp_new_fsurdat.2_21-i2000sp.2_21/set2/set2_ANN_TLAKE_0.png) , evident by the upper plots, but nothing big really stands out here.
- [x] Soil properties have changed quite a bit, which changed soil thermal and hydraulic properties, especially in the Arctic.  This makes summer soil conditions in the new simulations [warmer](https://webext.cgd.ucar.edu/I2000/i2000sp_new_fsurdat/lnd/i2000sp_new_fsurdat.2_21-i2000sp.2_21/set2/set2_JJA_TSOI_4.png) and [wetter](https://webext.cgd.ucar.edu/I2000/i2000sp_new_fsurdat/lnd/i2000sp_new_fsurdat.2_21-i2000sp.2_21/set2/set2_JJA_SOILLIQ_4.png), with less [ice](https://webext.cgd.ucar.edu/I2000/i2000sp_new_fsurdat/lnd/i2000sp_new_fsurdat.2_21-i2000sp.2_21/set2/set2_JJA_SOILICE_4.png) across the Arctic. 
   - @olyson will provide a look at ALTMAX 

- [x] @slevisconsulting will run parallel [BGC cases and post additional diagnostics](https://webext.cgd.ucar.edu/I20TR/ihist_bgccrop/lnd/ihist_bgccrop.1995_2014-ihist_bgccrop_old_fsurdat.1995_2014/setsIndex.html) and [ILAMB plots](https://webext.cgd.ucar.edu/I20TR/ihist_bgccrop/lnd/_build_fsurdat/)

**Comment by olyson** [comment]
ALTMAX is only output for BGC simulations.  So we'll have to use the BGC transient simulations Sam is running.

**Comment by slevis-lmwg** [comment]
> @slevisconsulting will run parallel BGC cases and post additional diagnostics

I have encountered a problem with the new transient bgc crop simulation; suggestions welcome:
- Case ihist_bgccrop with default datasets has run 1850-1973 this far and continues, so no problem here.
- The parallel case with the new datasets ran from 1850-1920 with datm cycling 1901-1920 BUT...
I have tried to continue this as a hybrid or a startup using restart files from 1921/1/1 or 1920/1/1 and I get a methane balance error.

The new datasets seem like the logical culprit, except for the fact that 1920 completed fine as a one-year restart in the cycling part of the simulation (1850-1920), while it triggers the methane balance error as a hybrid or startup for the next part (1920-2014). That's all the info I have for now. I will be troubleshooting.

**Reply by slevis-lmwg** [reply]
The transient case with the new datasets still gives me a methane balance error as follows:
- No problems running/restarting 1850-1920 with datm cycling 1901-1920.
- Hybrid or startup runs that I tried from 1901/1/1 or 1920/1/1 or 1921/1/1 gave methane balance errors.

I will continue troubleshooting. Suggestions are welcome.

**Comment by olyson** [comment]
During the initial development of dynamic urban we encountered a methane error.  See more details in doc/design/dynamic_urban.rst.  But this occurred the first time PCT_URBAN changed.  PCT_URBAN shouldn't change until year 2001 and we had fixed that error anyway.  So I don't think this is relevant, just wanted to make you aware of it.  I guess you could check to see what column type(s) are contributing to that gridcell error.

**Reply by slevis-lmwg** [reply]
Thank you, @olyson . Considering that you fixed the error for urban in dev069, now I suspect a similar problem with transient lakes. To confirm, I have submitted the hybrid case with `do_transient_lakes = .false.`. If that doesn't fail, then the problem resembles the urban one, though it arises from a difference between hybrid and restart (restarts work).

I haven't explicitly checked the column type(s) contributing to the error, but I've looked at the fsurdat file. The gridcell contains 46% lake (SW corner of Lake Michigan). It also has PCT_URBAN(3) = (0.044, 5.21, 42.25) and happens to be the main Chicago gridcell (x=218, y=140 in ncview).

**Reply by slevis-lmwg** [reply]
From conversation with @ekluzek and @billsacks at Standup:
Since restarts work, I may be able to continue the simulation past 1920 with a branch that ignores the datm restart file.

**Reply by slevis-lmwg** [reply]
@billsacks suggested this troubleshooting idea:
Try a hybrid start from a timeslice run in case this is not a transient problem.

**Reply by slevis-lmwg** [reply]
The `do_transient_lakes = .false.` test didn't start all night. Cheyenne seems busier than usual. So here's my new plan:
- I canceled that test for now.
- I submitted a branch where I removed the datm restart, as suggested two posts up. If this works, then we can at least proceed with the simulation.
- Meanwhile I'm setting myself up to run single-point at the Chicago gridcell that triggers the error, in order to accelerate the debugging.

Side note (I may have to open an issue for this): 
I tried `./subset_data point --lat 41.94 --lon 272.5 --create-surface --create-landuse --crop --include-nonveg` on my 1-degree fsurdat and landuse files. It generated the 1x1 fsurdat but it gave this error when attempting to generate the 1x1 landuse file:
```
  File "/glade/work/slevis/git/mksurfdata_toolchain/tools/site_and_regional/../../python/ctsm/subset_data.py", line 466, in subset_point
    args.user_mods_dir)
  File "/glade/work/slevis/git/mksurfdata_toolchain/tools/site_and_regional/../../python/ctsm/site_and_regional/single_point_case.py", line 367, in create_landuse_at_point
    f_out = f_out.transpose("time", "cft", "natpft", "lsmlat", "lsmlon")
  File "/glade/work/slevis/conda-envs/ctsm_py/lib/python3.7/site-packages/xarray/core/dataset.py", line 4093, in transpose
    "permuted dataset dimensions (%s)" % (dims, tuple(self.dims))
ValueError: arguments to transpose (('time', 'cft', 'natpft', 'lsmlat', 'lsmlon')) must be permuted dataset dimensions (('cft', 'lsmlat', 'lsmlon', 'natpft', 'numurbl', 'time'))
```
I will return to this if my workaround with mksurfdata_esmf fails...

**Reply by slevis-lmwg** [reply]
Good news:
- The branch (where I removed the datm restart) worked. The transient simulation 1921-2014 is now in progress.
- The subset_data failure that I reported had a trivial fix and works again. I will address this fix along with other small fixes in the ctsm5.2 branch (will open a PR).

One idea that didn't work from the previous post:  my workaround using mksurfdata_esmf. I haven't found a task geometry that will work for 1x1 and a task geometry that had worked in the past now fails.

**Reply by slevis-lmwg** [reply]
#43 describes a ch4 balance error with transient lakes. My ch4 balance error may be the same, although:
- I have yet to confirm that my problem is due to transient lakes and
- My balance error appears in hybrid/startup cases (e.g. 1901, 1920, 1921) though not in 1850 where the whole simulation started and not in restart/branch cases.

#43 suggests this workaround:
Comment out the gridcell balance check when running with transient lakes, maybe only for the first timestep of every year. The way the problem manifests for me, we may even just comment it out for the first timestep of the run.

**Reply by slevis-lmwg** [reply]
I plan to troubleshoot and address the ch4 balance check issue in #1873 
For now I see this code already in place in subr. ch4:
```
      ! Gricell level balance
      !

      ! Skip the check if it's the beginning of a new year and dynamic lakes are on
      ! See (https://github.com/ESCOMP/CTSM/issues/1356#issuecomment-905963583)
      !
      if ( is_beg_curr_year() .and. get_do_transient_lakes() )then
         ch4_first_time_grc(begg:endg) = .true.
      end if
```
So it seems likely that the error will go away if I add a comparable flag for the first timestep of hybrid/startup simulations. I will try that next.

**Reply by slevis-lmwg** [reply]
Hybrid and Startup cases that were giving the ch4 balance error are now working.
See https://github.com/ESCOMP/CTSM/pull/1873/commits/858539a6114e5b7dce4bb9db101ba9976b9e5411 in #1873

**Reply by billsacks** [reply]
Awesome, great work on this @slevisconsulting !

**Reply by slevis-lmwg** [reply]
Thanks @billsacks and also thank you and @ekluzek for making suggestions at last wk's Standup.

**Comment by slevis-lmwg** [comment]
> @slevisconsulting will run parallel BGC cases and post additional diagnostics

Diagnostics posted [here.](https://webext.cgd.ucar.edu/I20TR/ihist_bgccrop/lnd/ihist_bgccrop.1995_2014-ihist_bgccrop_old_fsurdat.1995_2014/setsIndex.html)

**Reply by slevis-lmwg** [reply]
In "line plots" the TOTRUNOFF difference plot may suggest some kind of problem a little before 1950.

**Comment by olyson** [comment]
I think we've noticed this before, it follows the GSWP3V1 precip trend.

**Comment by olyson** [comment]
ILAMB diagnostics here:

https://webext.cgd.ucar.edu/I20TR/ihist_bgccrop/lnd/_build_fsurdat/

**Comment by lawrencepj1** [comment]
@slevisconsulting Thanks for running and posting all of this. I found a pretty crazy spike in Landuse Flux in the diagnostics which I will look into.

![image](https://user-images.githubusercontent.com/11986105/196315543-b703639b-4f1a-471e-aff4-73f4a9177b72.png)

**Reply by slevis-lmwg** [reply]
Thank you @lawrencepj1

**Reply by wwieder** [reply]
From our meeting I think we identified this as an issue with the lakes dataset.

**Comment by olyson** [comment]
I've added a new soil moisture storage plot to set3.  
Also added ALTMAX to set2, hopefully with a better color scale.  See also the updated slides for ALTMAX:

https://docs.google.com/presentation/d/1fmReSb6XKQjChSjMTAxAp-p0BAR4HibsHya1y43udeg/edit#slide=id.g1630bf08395_0_25

The SP and BGC diagnostics pages have been updated with ALTMAX and soil moisture storage.

**Comment by wwieder** [comment]
@lawrencepj1 have you been able to make plots comparing soil datasets for CTSM5.1 and 5.2 simulations?

**Comment by lawrencepj1** [comment]
Sorry Will totally spaced on this. Will get on it today.
-- 
Dr Peter Lawrence
Terrestrial Science Section
National Center for Atmospheric Research
1850 Table Mesa Drive
Boulder Colorado 80305

Work: 1-303-497-1727
Cell: 1-303-956-6932

**Reply by wwieder** [reply]
Not to worry Peter.  I know you were involved with the workshop this week too, hope it went well.

**Reply by wwieder** [reply]
@lawrencepj1 have you been able to make plots from the first item in this discussion, or should someone else dive into this?  Especially related to differences in tree and grass cover with the [new surface dataset](https://webext.cgd.ucar.edu/I2000/i2000sp_new_fsurdat/lnd/i2000sp_new_fsurdat.2_21-i2000sp.2_21/set2/set2.html)?

![image](https://user-images.githubusercontent.com/8031012/204669793-08bd6152-f287-4014-a9f6-0382038d1a0e.png)

**Reply by lawrencepj1** [reply]
Hi Will

I just checked and while some of these changes in trees are due to the new lake dataset much of the changes shown in this plot are coming from updates to the raw pft/cft data from redoing the crop and pasture data. I have put in a PCT_TREE plot for the CTSM52 raw data vs the CLM5 data for the year 2015. For turn around time the plot is from ncview with the legend running over similar ranges as the ncl diagnostic difference plot.

![image](https://user-images.githubusercontent.com/11986105/205134584-f0eb1511-63eb-4a13-af97-9f1f2ed4cc4f.png)

**Reply by lawrencepj1** [reply]
@wwieder realized I didn't tag you on this update

**Reply by wwieder** [reply]
excellent. Thanks Peter.  Changes in China are really big!  Maybe we can post differences in soil properties here too for completeness?

**Comment by wwieder** [comment]
Hello, I wanted to follow up on a few points from our conversation on this a few weeks ago:

- [x] **Shifting cultivation** @lawrencepj1 this needs to be added?  What's an appropriate timeline to implement this?
- [x] **Lake dataset** @slevisconsulting were you going to investigate the lake area issues around 1940?
- [x] **Soil properties** @lawrencepj1  have you been able to compare 'ORGANIC' values in surface soils from the two surface data?
- [x] **Permafrost** @olyson made [new difference plots in ALT](https://docs.google.com/presentation/d/1fmReSb6XKQjChSjMTAxAp-p0BAR4HibsHya1y43udeg/edit?usp=sharing).  @dlawrenncar is this enough to evaluate permafrost dynamics?
- [ ] **Wetlands** @billsacks has opened a PR for this.

Are there other topics to follow up on?

**Reply by slevis-lmwg** [reply]
> * [ ]  **Lake dataset** @slevisconsulting were you going to investigate the lake area issues around 1940?

I was in touch with @Ivanderkelen roughly 10 days ago, and she agreed to send updated raw files for 1940 and 1941 soon.

> Are there other topics to follow up on?

- For completeness, I am expecting more new raw datasets from @lawrencepj1 
i.e. for 850-1849 and for the SSP/RCPs.
- Also from @lawrencepj1 I'm expecting new hi-res raw datasets, some for the old hi-res resolution and all for the new hi-res resolution. Then I'm supposed to test the hi-res option.
- @ekluzek is looking at #309 and then we are supposed to merge to the ctsm5.2 branch.

**Reply by slevis-lmwg** [reply]
> > * [ ]   **Lake dataset** @slevisconsulting were you going to investigate the lake area issues around 1940?
> 
> I was in touch with @Ivanderkelen roughly 10 days ago, and she agreed to send updated raw files for 1940 and 1941 soon.

@Ivanderkelen provided new raw files for 1940 and 1941 that look good at first glance.

**Reply by wwieder** [reply]
Where are we with PR #309 now?  From Sam's message above it looks like Erik's working on it?

Are there other datasets that still have to be merged?

**Reply by slevis-lmwg** [reply]
> Where are we with PR #309 now? From Sam's message above it looks like Erik's working on it?
> 
> Are there other datasets that still have to be merged?

@wwieder I didn't respond to this previously since the TODO list -- including new datasets -- comes from earlier in this Nov 3 post. However, I wanted to remind those involved of the pieces still missing:
- @ekluzek is supposed to look at #309 before reviewing with @slevisconsulting and merging; @ekluzek I'm available when you're ready.
- @lawrencepj1 has four TODOs listed:  Two involving raw data generation, one about comparing soil properties, and one about shifting cultivation. Peter, please see details earlier in this Nov 3 post. Could you provide an estimated completion date for each of these TODOs?
- Bill's PR #1920 has been reviewed by @slevisconsulting and is ready for review by @ekluzek, at which point it will be merged.

I think Peter may be at the AGU and soon after it's the holidays, so I don't mean to come across as pressuring. Just bringing this back up.

**Reply by ekluzek** [reply]
Peter is at AGU and then as you point out will be out for the holidays. So we'll get with him in the new year about his TODO's.

**Reply by ekluzek** [reply]
The shifting cultivation is something that's on me. There was a couple PR's about it and I believe we integrated one, but didn't the other as I thought it was tied up with CTSM5.2 dataset issues. But, I don't think that's the case, and I think that PR can just come in as it is. I need to look at it again and make sure that's the case.

**Reply by lawrencepj1** [reply]
Thanks Sam and Erik

Yes I am at AGU this week. I can look at the soil properties potentially next week. The shifting cultivation element is present in the new datasets. I will work on the PR with Erik. By default the shifting cultivation is inactive so this should not be a major issue for CTSM5.2 release. 

Cheers
Peter

**Reply by slevis-lmwg** [reply]
Notes regarding the above from 2023/1/5 meeting with @lawrencepj1 @jkshuman @ekluzek @slevisconsulting:
- @lawrencepj1 will generate the new 850-2100 pft data plus lai and soilcolor data this month. He needs a new mesh file to complete this work; @lawrencepj1 will reach out to @ekluzek (or @slevisconsulting ) in the next week or so to generate the new mesh file.
- @lawrencepj1 will generate corresponding timeslice data for the standard high-res resolution. Decision is pending whether to offer another high-res resolution and whether to offer timeseries data for the high-res resolution(s).
- @lawrencepj1 will place new files in the `/glade/p/cesmdata/cseg/inputdata/lnd/clm2/rawdata` space; @slevisconsulting (or @ekluzek ) will ./rimport the files to the svn repo.
- @lawrencepj1 will move his codes/workflow to github.com/ESCOMP and will teach others (@slevisconsulting @ekluzek ...) how to use them.
- @slevisconsulting will do this one: soil properties comparison.
- @ekluzek will coordinate a mtg for a post-ctsm5.2 discussion (Mon or Tues next wk?).

Corrections or comments on the above TODO list are welcome.

**Reply by wwieder** [reply]
thanks for posting these notes, @slevisconsulting.

Was timeline for this discussed?

**Reply by lawrencepj1** [reply]
Hi Will

Yes we talked about the timelines for the first two component being the end of next week for 1. and by the end of the week after for 2. The remaining tasks will be over the next month to two months. We will be having a follow up meeting next week to discuss post CTSM5.2 issues. This will include the FATES / LUH2(3) developments required to include primary and secondary forests and non-forests as well as pasture and rangeland. It will also include the ability to represent some data currently on the surface and landuse timeseries files on streams. This may involve updating the current climatology LAI/SAI/Canopy height data along with harvest and fertilizer data.

Peter 

components 1 and 2:

@lawrencepj1 will generate the new 850-2100 pft data plus lai and soilcolor data this month. He needs a new mesh file to complete this work; @lawrencepj1 will reach out to @ekluzek (or @slevisconsulting ) in the next week or so to generate the new mesh file.
@lawrencepj1 will generate corresponding timeslice data for the standard high-res resolution. Decision is pending whether to offer another high-res resolution and whether to offer timeseries data for the high-res resolution(s).

**Reply by wwieder** [reply]
Thanks for these additional details.

I see from @slevisconsulting's notes above that @ekluzek is also supposed to organize a post CTSM5.2 discussion.  Is this something we should do at our weekly CLM meeting on Thursday, or does the conversation need a smaller, more focused group?

**Reply by slevis-lmwg** [reply]
@wwieder 
Erik will set up the post-ctsm5.2 discussion most likely for 1/10 (tomorrow).

**Reply by slevis-lmwg** [reply]
Update of TODOs in this discussion:

> * @lawrencepj1 will generate the new 850-2100 pft data plus lai and soilcolor data this month.

DONE out to 2300

> * @lawrencepj1 will generate corresponding timeslice data for the standard high-res resolution.

**PEND**

> * @lawrencepj1 will place new files in the `/glade/p/cesmdata/cseg/inputdata/lnd/clm2/rawdata` space

DONE for 850-2300
**PEND for high-res**

> @slevisconsulting (or @ekluzek ) will ./rimport the files to the svn repo

**TODO after testing** as part of PR #1946

> * @lawrencepj1 will move his codes/workflow to github.com/ESCOMP and will teach others (@slevisconsulting @ekluzek ...) how to use them

**PEND**

> @ekluzek is supposed to look at #309 before reviewing with @slevisconsulting and merging

@slevisconsulting will update #309 to main and perform testing; @ekluzek will merge when ready
#1920 got merged

I generated the 1.9x2.5 transient surface data using the new raw pft data as requested by @samsrabin. SamR is testing. I posted a bit more detail in #1946.

**Reply by lawrencepj1** [reply]
Thanks for the update on the pending and completed tasks @slevisconsulting

**Comment by dlawrenncar** [comment]
I think the permafrost simulation is fine.  There are some changes, but
nothing so big that it would be a big concern.

**Comment by slevis-lmwg** [comment]
New minus old ORGANIC [kg m-3]. The video starts with layer 1 and loops through the 10 soil layers multiple times.

https://user-images.githubusercontent.com/35322900/212502614-a29304c2-5313-45d3-bc4e-2bed04d72df8.mp4

![org_diff_colorbar](https://user-images.githubusercontent.com/35322900/212502793-ce36ac56-85a9-42ab-80c7-fe9ea4a324bf.JPG)

**Reply by slevis-lmwg** [reply]
New ORGANIC [kg m-3]. The video starts with layer 1 and loops through the 10 soil layers multiple times.

https://user-images.githubusercontent.com/35322900/212503067-2aac5a4e-32fb-4862-a182-9b6601c4f741.mp4

![org_new_colorbar](https://user-images.githubusercontent.com/35322900/212503066-1839b49b-6499-4e27-9bcd-e2b77276c018.JPG)

**Reply by slevis-lmwg** [reply]
Old ORGANIC [kg m-3]. Same colorbar as the new (directly above).
The video starts with layer 1 and loops through the 10 soil layers multiple times. The old organic goes to 0 everywhere in the bottom two layers.


https://user-images.githubusercontent.com/35322900/212503355-6eadc4a4-3c33-4e2a-8301-efb19f98bdf0.mp4

**Comment by slevis-lmwg** [comment]
New minus old SAND [-50% to 50%]. The video starts with layer 1 and loops through the 10 soil layers multiple times.


https://user-images.githubusercontent.com/35322900/212503916-0d9c3de4-f151-4606-b011-7ebff85316f9.mp4

![sand_or_clay_diff_colorbar](https://user-images.githubusercontent.com/35322900/212751270-ba484a9f-ed35-497a-8b92-d3bbce5c84ac.JPG)

**Reply by slevis-lmwg** [reply]
New SAND [0 to 100%]. The video starts with layer 1 and loops through the 10 soil layers multiple times.



https://user-images.githubusercontent.com/35322900/212504526-ff53528b-20f9-41f1-a860-637f3d88f859.mp4


![sand_or_clay_colorbar](https://user-images.githubusercontent.com/35322900/212751535-352666ec-3b35-435e-adff-c2d83a162d4b.JPG)

**Reply by slevis-lmwg** [reply]
Old SAND [0 to 100%]. Same colorbar as the new (directly above). The video starts with layer 1 and loops through the 10 soil layers multiple times.



https://user-images.githubusercontent.com/35322900/212504535-22febb9a-28be-43f1-8223-3fed74103e5c.mp4

**Comment by slevis-lmwg** [comment]
New minus old CLAY [-50% to 50%]. The video starts with layer 1 and loops through the 10 soil layers multiple times.


https://user-images.githubusercontent.com/35322900/212750745-539f5538-1513-45a4-8ae9-5c8f8cb2140a.mp4

![sand_or_clay_diff_colorbar](https://user-images.githubusercontent.com/35322900/212752017-9e73e122-d420-4380-8f70-e831a1613545.JPG)

**Reply by slevis-lmwg** [reply]
New CLAY [0 to 100%]. The video starts with layer 1 and loops through the 10 soil layers multiple times.


https://user-images.githubusercontent.com/35322900/212750867-3030cc01-3752-4f88-9152-965ac8eb92ba.mp4

![sand_or_clay_colorbar](https://user-images.githubusercontent.com/35322900/212752072-06f2f224-2b2f-48ed-93b6-ba139836e144.JPG)

**Reply by slevis-lmwg** [reply]
Old CLAY [0 to 100%]. Same colorbar as the new (directly above). The video starts with layer 1 and loops through the 10 soil layers multiple times.


https://user-images.githubusercontent.com/35322900/212750964-2f524576-2d41-411e-ad93-1b7ed81756ee.mp4

**Comment by slevis-lmwg** [comment]
New minus old HKSAT [-0.27 to 0.27 mm s-1]. Same colorbar colorscheme as prev. variables. The video starts with layer 1 and loops through the 25 ground levels a few times.


https://user-images.githubusercontent.com/35322900/212760821-59a1bbe4-2b3f-45e2-87fb-fc271561cd8f.mp4

**Reply by slevis-lmwg** [reply]
New HKSAT [0 to 0.2744 mm s-1]. Same colorbar colorscheme as prev. variables. The video starts with layer 1 and loops through the 25 ground levels a few times.


https://user-images.githubusercontent.com/35322900/212761450-c4dbd56c-5c83-4e3e-9e84-770118a63031.mp4

**Reply by slevis-lmwg** [reply]
Old HKSAT [0 to 0.2744 mm s-1]. Same colorbar colorscheme as prev. variables. The video starts with layer 1 and loops through the 25 ground levels a few times.


https://user-images.githubusercontent.com/35322900/212761651-60c56ed2-8258-475d-ad75-fba74e658734.mp4

**Comment by wwieder** [comment]
At the Feb 28th co-chairs meeting we're discussing developments in component models with an eye to the next round of coupled model testing that's scheduled to start in March.  Can we consider using a 5.2 surface dataset on this timeframe (just for a single model resolution), or would this be too soon to be ready?  My guess is too soon, as we also need to provide initial conditions for coupled model runs, but I thought I'd ask.

**Reply by samsrabin** [reply]
Depending on when the revised 5.2 datasets become available at "2-degree" resolution, I may rerun the experiments for my crop calendar paper with them. This will include a "CLM Default" run starting with the 1850 surface dataset, then time-varying land use. 

That said, I'm not set up to do analysis of anything other than crop-related outputs.

**Reply by lawrencepj1** [reply]
Hi @slevisconsulting and @samsrabin Yes the timeframe for the datasets should be fine. The issues on crop distribution we found with Sam Rabin have been resolved so the raw data will be updated in the next week for the February 28th meeting. Thanks Peter

**Reply by slevis-lmwg** [reply]
A missing piece still is #309 which is close but not done. Assuming today's test-suite passes, I have two more updates before we merge to main, update the ctsm5.2 branch, and test again. Other than waiting for test-suites, the slowest part of the process is debugging conflicts that don't get caught by git.

Having said all that, I think it makes sense to generate the needed fsurdat now, and then again when #309 is in. I could do that if you let me know what resolution we need for the coupled testing.

**Reply by wwieder** [reply]
Thanks @slevisconsulting I'll start an email thread with those involved in the coupled model tests to ask about resolution.  I like the idea of providing a beta version of a 5.2 surface dataset to testing.  

How time consuming is it going to be to generate initial conditions to run in the coupled model, @olyson?

Is it worth this effort for testing coupled configurations?  (I'm pretty naive regarding how to proceed with these coupled model tests).

**Comment by olyson** [comment]
As you suggested, we should have a science discussion on this.  I think there were some discussions about spinning up CLM using forcing from a CAM-CLM F-case simulation in which CAM and CLM are as close to the fully coupled configuration being used for CESM development simulations.  We used to do this type of spinup using coupler history files but I thought I saw somewhere that nuopc didn't support that kind of configuration?  So there may be a bit of time involved in understanding how to do this.

**Reply by billsacks** [reply]
> I saw somewhere that nuopc didn't support that kind of configuration? So there may be a bit of time involved in understanding how to do this.

I believe this will be possible once we integrate https://github.com/ESCOMP/CTSM/pull/1929

**Reply by billsacks** [reply]
Actually, I think my last comment may have been wrong. @adrifoster reminded me of #1844 , and I think that may be the remaining problem. If so, I'm not sure that #1929 is strictly needed for that. (I'm having deja vu: I know we've gone around a few times about what #1929 is and isn't actually needed for.) On the other hand, I think that https://github.com/ESCOMP/CAM/pull/735 *is* needed: Once CAM always passes NDEP, we can generate new cplhist files from nuopc/cmeps, then use those new cplhist files to force datm using CDEPS. At that point, it would probably make sense to also change CTSM to actually listen to the NDEP sent from ATM. (The need for the CAM update is because DATM expects NDEP on its forcing files, even though CTSM doesn't actually use NDEP from ATM yet.)

**Reply by billsacks** [reply]
See https://github.com/ESCOMP/CTSM/discussions/1951#discussioncomment-4989269 for more details.

**Reply by klindsay28** [reply]
> change CTSM to actually listen to the NDEP sent from ATM.

I thought that was already happening. Is it not done by the following block in the NUOPC cap?https://github.com/ESCOMP/CTSM/blob/05fb17bff05aaf0544170ecbb41984bdbb875807/src/cpl/nuopc/lnd_import_export.F90#L668-L677

**Comment by olyson** [comment]
Ok, sorry for mis-stating this, I was thinking of this discussion:

https://github.com/NCAR/amwg_dev/issues/188

in which Adam and Mariana were debugging the cpl history output option in nuopc.
The last entry states that "Quick update - we've resolved the issue of generating the streams w/ nuopc. We are meeting on wednesday to iron out setting those streams up for forcing clm standalone."

I'm not sure if this is associated with an issue/pull request that has been implemented already.

**Comment by billsacks** [comment]
@klindsay28 and @olyson - sorry, I feel like my comments have been more misleading / wrong than helpful: they were based on my recollections of conversations from a few months ago rather than any first-hand knowledge of the current state. I think you both have a better sense of the current state of affairs than I do.

**Comment by wwieder** [comment]
@olyson completed a historical using the datasets @slevisconsulting created. Diagnostics compared to a CLM5 run are here:

https://webext.cgd.ucar.edu/I20TR/ctsm51_ctsm51d118_ne30pg3g17_GSWP3V1_hist/lnd/ctsm51_ctsm51d118_ne30pg3g17_GSWP3V1_hist.1994_2013-clm50_ctsm51d118_ne30pg3g17_GSWP3V1_hist.1994_2013/setsIndex.html

I don't see anything unexpected. For example, the spike in the landuse flux that @lawrencepj1 noted previously (due to a lake dataset problem) isn't there.

**Comment by lawrencepj1** [comment]
Hi @slevisconsulting @wwieder @ekluzek @samsrabin 

Just making a quick note that the distributions of individual CFTs has been updated following suggestions from Sam Rabin and as presented at the 2023 LMWG meeting. The updated files are now available on 

/glade/p/cesmdata/cseg/inputdata/lnd/clm2/rawdata/

pftcftdynharv.0.25x0.25.LUH2.histsimyr0850-1850.c20230226
pftcftdynharv.0.25x0.25.LUH2.histsimyr1850-2015.c20230226
 
pftcftdynharv.0.25x0.25.SSP1-1.9.simyr2015-2100.c20230226  
pftcftdynharv.0.25x0.25.SSP1-2.6.simyr2015-2100.c20230226
pftcftdynharv.0.25x0.25.SSP2-4.5.simyr2015-2100.c20230226  
pftcftdynharv.0.25x0.25.SSP3-7.0.simyr2015-2100.c20230226  
pftcftdynharv.0.25x0.25.SSP4-3.4.simyr2015-2100.c20230226  
pftcftdynharv.0.25x0.25.SSP4-6.0.simyr2015-2100.c20230226
pftcftdynharv.0.25x0.25.SSP5-3.4.simyr2015-2100.c20230226
pftcftdynharv.0.25x0.25.SSP5-8.5.simyr2015-2100.c20230226
 
pftcftdynharv.0.25x0.25.SSP1-2.6.simyr2100-2300.c20230226  
pftcftdynharv.0.25x0.25.SSP5-3.4.simyr2100-2300.c20230226
pftcftdynharv.0.25x0.25.SSP5-8.5.simyr2100-2300.c20230226

Note the directories and file names now for 2023 02 26.

These updated raw files should be considered the CTSM 5.2 land cover and land use data rather than the previous version. We should remove the 2023 01 16 version once we have this finalized.

Thanks
Peter

**Reply by slevis-lmwg** [reply]
Thank you @lawrencepj1

**Comment by samsrabin** [comment]
My f19 crop runs with the 5.2 datasets are complete, and I've made some simple global timeseries figures to illustrate the difference. Overall, area looks much better, and the production (Mt) values look either improved or similar. Yield (t/ha) is either similar or slightly worsened (overestimate increased, for corn and wheat). 

Here, cyan is with the 5.0 datasets and pink is with 5.2. "Total (grains)" means everything except cotton and sugarcane.

### Area
![Timeseries Global crop area by crop](https://user-images.githubusercontent.com/10454527/223222562-d802cf4c-1e5f-4f0b-8798-95643bda7a04.png)

### Production
![Timeseries Global crop production by crop](https://user-images.githubusercontent.com/10454527/223222620-a44e94b2-5a49-47ae-8508-c95082c53410.png)

### Yield
Numbers in subplot titles are mean bias over 1980–2009, 5.0 → 5.2.
![Timeseries Global crop yield by crop](https://user-images.githubusercontent.com/10454527/223222643-27e6e4c2-4c73-4433-bd81-b18ab25a4b81.png)

**Comment by ekluzek** [comment]
Closing as this is now old, we've moved to ctsm5.3 datasets with ctsm5.4 being close as well.
